File: backend/entity/payment.go
// entity/payment.go
package entity

import (
	"time"

	"gorm.io/gorm"
)

type Payment struct {
	gorm.Model
	PaymentDate time.Time `json:"payment_date"`
	Status      string    `json:"status"`
	Amount      float64   `json:"amount"`

	OrderID uint   `json:"order_id"`
	Order   *Order `gorm:"foreignKey:OrderID" json:"order,omitempty"`

	PaymentSlips []PaymentSlip `gorm:"foreignKey:PaymentID" json:"payment_slips,omitempty"`
}

File: frontend/src/interfaces/Payment.ts
import type { Order } from "./Order";
import type { PaymentSlip } from "./PaymentSlip";

export interface Payment {
  ID: number;
  payment_date: string; // ISO datetime
  status: string;       // e.g., "pending", "verifying", "approved", "rejected"
  amount: number;

  order_id: number;
  order?: Order;

  payment_slips?: PaymentSlip[];
}

export interface CreatePaymentRequest {
  payment_date?: string;
  status: string;
  amount: number;
  order_id: number;
}

export interface UpdatePaymentRequest {
  ID: number;
  payment_date?: string;
  status?: string;
  amount?: number;
}
